hl7-helper/hl7-parser/src/main/java/gov/cdc/dataingestion/hl7/helper/integration/HL7Parser.java

```diff
@@ -0,0 +1,110 @@
+package gov.cdc.dataingestion.hl7.helper.integration;
+
+import ca.uhn.hl7v2.HL7Exception;
+import ca.uhn.hl7v2.HapiContext;
+import ca.uhn.hl7v2.model.Message;
+import ca.uhn.hl7v2.parser.CanonicalModelClassFactory;
+import ca.uhn.hl7v2.parser.DefaultXMLParser;
+import ca.uhn.hl7v2.parser.PipeParser;
+import ca.uhn.hl7v2.parser.XMLParser;
+import ca.uhn.hl7v2.util.Terser;
+import ca.uhn.hl7v2.validation.impl.ValidationContextFactory;
+import gov.cdc.dataingestion.hl7.helper.integration.interfaces.IHL7Parser;
+import gov.cdc.dataingestion.hl7.helper.model.HL7ParsedMessage;
+import gov.cdc.dataingestion.hl7.helper.model.PatientAddress;
+import gov.cdc.dataingestion.hl7.helper.model.PatientIdentification;
+import gov.cdc.dataingestion.hl7.helper.model.PatientName;
+
+public class HL7Parser implements IHL7Parser {
+
+    private HapiContext context;
+    private final String newLine = "\n";
+    private final String carrier = "\r";
+
+    // this is the support hl7 structure
+    private final String supportedHL7version = "2.5.1";
+
+    public HL7Parser(HapiContext context) {
+        this.context = context;
+    }
+
+    public String hl7MessageStringValidation(String message) throws DiHL7Exception {
+        if(message.contains(newLine)) {
+            message = message.replaceAll(newLine, carrier);
+        } else if (!message.contains(carrier)) {
+            throw new DiHL7Exception("Incorrect raw message format");
+        }
+        return message;
+    }
+
+    public HL7ParsedMessage hl7StringParser(String message) throws DiHL7Exception {
+
+        try {
+            context.setValidationContext(ValidationContextFactory.defaultValidation());
+            PipeParser parser = context.getPipeParser();
+            Message parsedMessage = parser.parse(message);
+
+            Terser terser = new Terser(parsedMessage);
+
+            String messageType = terser.get("/MSH-9-1");
+            String messageEventTrigger = terser.get("/MSH-9-2");
+            String messageVersion = parsedMessage.getVersion();
+
+            PatientName patientName = new PatientName();
+            PatientAddress patientAddress = new PatientAddress();
+            PatientIdentification patientIdentification = new PatientIdentification();
+
+            patientName.setFamilyName(terser.get("/PID-5-1"));
```

Comment:
This is how we get data from hl7 message without needing to cast the message type
But, we need to know the exact location of the info we want to extract